src/Worker.Extensions.DurableTask/FunctionsOrchestrationContext.cs

```diff
@@ -0,0 +1,152 @@
+
+using System;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DurableTask;
+using Microsoft.DurableTask.Worker;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.DurableTask;
+
+internal sealed partial class FunctionsOrchestrationContext : TaskOrchestrationContext
+{
+    private readonly TaskOrchestrationContext innerContext;
+    private readonly FunctionContext functionContext;
+
+    private readonly DurableTaskWorkerOptions options;
+
+    private InputConverter? inputConverter;
+
+    public FunctionsOrchestrationContext(TaskOrchestrationContext innerContext, FunctionContext functionContext)
+    {
+        this.innerContext = innerContext;
+        this.functionContext = functionContext;
+        this.options = this.functionContext.InstanceServices
+            .GetRequiredService<IOptions<DurableTaskWorkerOptions>>().Value;
+    }
+
+    public bool IsAccessed { get; private set; }
+
+    public override TaskName Name => this.innerContext.Name;
+
+    public override string InstanceId => this.innerContext.InstanceId;
+
+    public override DateTime CurrentUtcDateTime => this.innerContext.CurrentUtcDateTime;
+
+    public override bool IsReplaying => this.innerContext.IsReplaying;
+
+    public override ParentOrchestrationInstance? Parent => this.innerContext.Parent;
+
+    public override T GetInput<T>()
+    {
+        this.EnsureLegalAccess();
+
+        object? input = this.innerContext.GetInput<object>();
+        if (input is T typed)
+        {
+            return typed;
+        }
+
+        // The wrapped TaskOrchestrationContext is not actually dynamic with GetInput - it was set
+        // once based on the declared input type of the orchestrator. Since we do not know the
+        // desired input type upfront, we were initialized to object. So we must serialize and
+        // deserialize again to convert to our desired type.
+        this.inputConverter ??= InputConverter.Create(input, this.options.DataConverter);
+        return this.inputConverter.Get<T>();
+    }
+
+    public override Guid NewGuid()
+    {
+        this.EnsureLegalAccess();
+        return this.innerContext.NewGuid();
+    }
+
+    public override Task<T> CallActivityAsync<T>(TaskName name, object? input = null, TaskOptions? options = null)
+    {
+        this.EnsureLegalAccess();
+        return this.innerContext.CallActivityAsync<T>(name, input, options);
+    }
+
+    [Obsolete("Not yet supported")]
+    public override Task<T> CallActivityAsync<T>(
+        Func<object?, T> activityLambda,
+        object? input = null,
+        TaskOptions? options = null)
+    {
+        this.EnsureLegalAccess();
+        return this.innerContext.CallActivityAsync(activityLambda, input, options);
+    }
```

Comment:
We should probably remove this. I'm not sure why I checked in it (might have been on accident).

Reply:
This is an `abstract` method, so I need some implementation. Are you proposing this is removed from the base type? If so, that won't make next preview.